File: pages/api/songs.py
import json
import requests

BACKEND_URL = "https://song-recommeder-backend-hbyp.onrender.com"

def handler(request):
    if request.method == "GET":
        query = request.args.get("q", "")
        if not query.strip():
            return {
                "statusCode": 200,
                "headers": {"Content-Type": "application/json"},
                "body": json.dumps([])
            }

        try:
            response = requests.get(f"{BACKEND_URL}/search", params={"q": query})
            response.raise_for_status()
            songs = response.json()
            return {
                "statusCode": 200,
                "headers": {"Content-Type": "application/json"},
                "body": json.dumps(songs)
            }
        except Exception as e:
            return {
                "statusCode": 500,
                "headers": {"Content-Type": "application/json"},
                "body": json.dumps({"error": str(e)})
            }

    elif request.method == "POST":
        try:
            body = request.get_json(silent=True) or {}
            response = requests.post(f"{BACKEND_URL}/recommend", json=body, headers={"Content-Type": "application/json"})
            response.raise_for_status()
            recs = response.json()
            return {
                "statusCode": 200,
                "headers": {"Content-Type": "application/json"},
                "body": json.dumps(recs)
            }
        except Exception as e:
            return {
                "statusCode": 500,
                "headers": {"Content-Type": "application/json"},
                "body": json.dumps({"error": str(e)})
            }

    else:
        return {
            "statusCode": 405,
            "headers": {"Content-Type": "application/json"},
            "body": json.dumps({"error": "Method not allowed"})
        }

File: pages/api/songs.ts
import { NextApiRequest, NextApiResponse } from "next";
import axios from "axios";

const BACKEND_URL = "https://song-recommeder-backend-hbyp.onrender.com";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  if (req.method === "GET") {
    const query = (req.query.q as string) || "";

    if (query.trim().length === 0) {
      return res.status(200).json([]);
    }

    try {
      const response = await axios.get(`${BACKEND_URL}/search`, {
        params: { q: query },
      });
      return res.status(200).json(response.data);
    } catch (error: any) {
      console.error("Backend error:", error.response?.data || error.message);
      return res.status(500).json({
        error: "Error fetching suggestions",
        details: error.response?.data || error.message,
      });
    }
  } else if (req.method === "POST") {
    try {
      const response = await axios.post(`${BACKEND_URL}/recommend`, req.body, {
        headers: {
          "Content-Type": "application/json",
        },
      });
      return res.status(200).json(response.data);
    } catch (error: any) {
      console.error("Backend error:", error.response?.data || error.message);
      return res.status(500).json({
        error: "Error getting recommendations",
        details: error.response?.data || error.message,
      });
    }
  } else {
    res.status(405).json({ error: "Method not allowed" });
  }
}
